src/metabase/notification/payload/impl/card.clj

```diff
@@ -11,11 +11,15 @@
 
 (mu/defmethod notification.payload/payload :notification/card
   [{:keys [creator_id payload subscriptions] :as _notification-info} :- ::notification.payload/Notification]
-  (let [card_id (:card_id payload)]
-    {:card_part       (notification.execute/execute-card creator_id card_id
-                                                         ;; for query_execution's context purposes
-                                                         ;; TODO: check whether we can remove this or rename it?
-                                                         :pulse-id (:id payload))
+  (let [card_id     (:card_id payload)
+        part        (notification.execute/execute-card creator_id card_id)
+        card-result (:result part)]
+    (when (not= :completed (:status card-result))
+      (throw (ex-info (format "Failed to execute card with error: %s" (:error card-result))
```

Comment:
I wonder if it possible we introduce other return shapes, which are not completed, but don't have an error either. 

I would be more comfortable if we handled known error status(es) explicitly, and threw a different exception with the full result if the status is something unexpected.

Reply:
wdyt about my idea here; https://github.com/metabase/metabase/pull/54949#discussion_r1988778496?

this is something I plan for the "proper error handling" project.

Reply:
Yeah that sounds good, push exceptions as far to the boundary as possible. Have clear data-driven functional control flow as much as possible.